File: 0107-binary-tree-level-order-traversal-ii/0107-binary-tree-level-order-traversal-ii.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    void traverse(vector<vector<int>>& res, TreeNode* node, int depth){
        if (node == NULL) {
            return;
        }

        if (res.size() <= depth) {
            vector<int> list;
            res.push_back(list);
        } 

        res[depth].push_back(node->val);

        traverse(res, node->left, depth + 1);
        traverse(res, node->right, depth + 1);
    }
    vector<vector<int>> levelOrderBottom(TreeNode* root) {
        vector<vector<int>> res;
        if (root == NULL) {
            return res;
        }

        traverse(res, root, 0);
        reverse(res.begin(), res.end());
        return res;
    }
};

File: 0107-binary-tree-level-order-traversal-ii/0107-binary-tree-level-order-traversal-ii.py
from collections import deque
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def levelOrderBottom(self, root: Optional[TreeNode]) -> List[List[int]]:
        res = []
        
        def traverse(node, level):
            if not node:
                return
            
            if len(res) < level + 1:
                res.append([])
            
            res[level].append(node.val)
            
            traverse(node.left, level + 1)
            traverse(node.right, level + 1)
            
        traverse(root, 0)
        
        res.reverse()
        
        return res
        
            